### ISSUE TITLE
Produce 2D Histograms, and then Violin Plots

### ISSUE THREAD

**Issue by lawrenceleejr** [lawrenceleejr/FragmentationStudy#19]
@cbell1402 will create root files with histograms for:

* 1 TeV Z+gamma
* 4 TeV Z+gamma
* 1 TeV JJJ
* 4 TeV JJJ
* 1 TeV ZZ
* 500 GeV ZZ

And then @cbell1402 will work with @jslawless to produce split violin plots.

**Comment by cbell1402** [comment]
The root files have been created and added [here](https://github.com/lawrenceleejr/FragmentationStudy/tree/dev/charles/root_files/violin_files). 
For some reason, eeToZgamma_50k_4TeV has few events. But the 1TeV version is fine. Any ideas why?

**Comment by lawrenceleejr** [comment]
Yeah I think this is for the reason we said last week. That in the pT range we're looking at, the only way to get jets there is to have them be soft radiation, so they're suppressed by a large factor. If you look at higher pT, the bulk of the jets are just up there.

**Comment by jslawless** [comment]
![eeToJJJ_50k_1TeV_n90](https://user-images.githubusercontent.com/116592104/217673266-7bf378cc-693a-4310-922a-a43986560e22.png)
![eeToJJJ_50k_4TeV_n90](https://user-images.githubusercontent.com/116592104/217673283-47f6efc5-6b62-495a-8c1c-263b32789d22.png)
![eeToZgamma_50k_1TeV_n90](https://user-images.githubusercontent.com/116592104/217673292-ffdecadf-310a-4406-ba47-1c79bb65ba39.png)
![eeToZgamma_50k_4TeV_n90](https://user-images.githubusercontent.com/116592104/217673303-046a958d-aef7-4c90-a024-6503702828d1.png)
![eeToZZToJJ_50k_1TeV_n90](https://user-images.githubusercontent.com/116592104/217673313-ce182927-fc71-43cc-924a-0f437f51532b.png)
![eeToZZToJJ_50k_500GeV_n90](https://user-images.githubusercontent.com/116592104/217673322-5f80e38a-9e94-402b-9f6d-f7f48e62756c.png)

**Comment by jslawless** [comment]
Does this look about expected?

**Comment by lawrenceleejr** [comment]
Thanks @jslawless! It's... unfortunately a bit hard to say. The color single cases do seem a bit flatter by eye, but I guess we do need more precise plots to really compare carefully (instead of me holding up a piece of paper to my screen). I'm hoping the n50 will look a little more different. But the split violins (and then adding in some profile plots on top of them) will be important to see any differences.

I just hacked together a split violin plot in keynote:

<img width="582" alt="Screen Shot 2023-02-09 at 9 30 07 AM" src="https://user-images.githubusercontent.com/7438278/217841192-f9e8882f-5560-4ff8-a1cf-d79d48106bbe.png">

This is between the 4 TeV ZGamma versus the 1 TeV JJJ (I think the 4 TeV JJJ would have looked similar). So that looks pretty good!! I think we have the histograms we need and now's just a game of making the plots look nice and really sell the message.

@jslawless if you want to drop a zip file or tarball or something of all the plots here, that would be helpful so we can play around while we also work on the split violins and whatever profile info we want to add at the end.

**Comment by jslawless** [comment]
[plots.zip](https://github.com/lawrenceleejr/FragmentationStudy/files/10700010/plots.zip)

**Comment by jslawless** [comment]
There's all my plots. I'm close to making the split violins, but a few hiccups are left.

**Comment by jslawless** [comment]
Missing some axes and axes values, but got a split violin plot working. This is for the N80 case.
![image](https://user-images.githubusercontent.com/116592104/217955196-c7046e66-8a03-4ee6-8c99-3b73d23fe07d.png)

**Comment by lawrenceleejr** [comment]
Just for our notes, the above plot is 4 TeV ee collisions for both processes. This looks great. Johnny is investigating drawing the behavior of the means to make it a bit easier to see the trends. Can you also try playing with the cut option to extend those vertical lines farther? And cut off the y axis at 0 since negative is unphysical.

**Comment by lawrenceleejr** [comment]
We can also then set inner to none.

**Comment by lawrenceleejr** [comment]
Consolidating to an actual task list for the drawing style because I was on my phone last night for the above messages.

* Change axis labels and titles
* Cut off y axis at 0, and probably also 15 or something on the high end. 
* Try playing with the `cut` parameter to make the vertical lines span the full y range of the plot.
* Draw two `plot` objects that show how the means change along the x axis, using colors that match the violin colors.
  * Might want to play with having a nice shadow for the profiles to suggest depth in the plot so the mean trends step forward and the full distributions recede a little. Would require a little playing around it looks like. Lower priority, but will be nice for the final thing.
* Remove the `box` portion in the middle. That can be done by setting `inner` to `None` it looks like.

**Comment by jslawless** [comment]
![Screenshot 2023-02-10 at 11 04 11 AM](https://user-images.githubusercontent.com/116592104/218138529-95d4a7dc-9958-49da-942d-a1ae196a78e3.png)
Got some of the more basic things done. I'll start playing around with the more advanced stuff.

**Comment by lawrenceleejr** [comment]
Nice. Updating todo list adding and removing some things:

* Change axis labels and titles. Can the x axis labels actually show the range? Or the bin end maybe? So maybe the first one says 0, and the next one says the start of that bin. Or we could do "0-25", but that might get too verbose. Play around with it. We should also make it "Jet P [GeV]". And the y axis, symbolically, let's call this $n_{80}$. You can do this in python/mpl as `r"$n_{80}$"`.
* Try playing with the cut parameter to make the vertical lines span the full y range of the plot.
* Draw two plot objects that show how the means change along the x axis, using colors that match the violin colors.
  * Might want to play with having a nice shadow for the profiles to suggest depth in the plot so the mean trends step forward and the full distributions recede a little. Would require a little playing around it looks like. Lower priority, but will be nice for the final thing.
* When convenient, instead of having a legend, I like to more directly label the data. (This is one of the lessons Tufte has.) Maybe like something below where it still clearly communicates the color coding, but also encodes the spatial system and does so directly on the data instead of in a separate key.
 
<img width="499" alt="Screen Shot 2023-02-10 at 11 13 22 AM" src="https://user-images.githubusercontent.com/7438278/218140741-5f450987-4008-4513-9ddf-dfd1ad3f3296.png">

**Comment by lawrenceleejr** [comment]
Comparisons for: {n50, n80, n90}

4 TeV JJJ vs (4 TeV) Z+gamma (already above for n80)
1 TeV JJJ vs (1 TeV) ZZ

@cbell1402 can you:
* Commit your latest code with the nXX plot creation?
* Can you change the binning in your histograms such that the number of y bins is 100.
* And for the n50 plots, can you set the y upper limit to 10 (from 20)

And we can focus on the samples needed for the two comparisons at the top of this message.

**Comment by cbell1402** [comment]
Updated loop.py [here](https://github.com/lawrenceleejr/FragmentationStudy/blob/dev/charles/loop.py).
I added the new root files [here](https://github.com/lawrenceleejr/FragmentationStudy/tree/dev/charles/root_files/violin_files).

**Comment by enibigir** [comment]
> Comparisons for: {n50, n80, n90}
> 
> 4 TeV JJJ vs (4 TeV) Z+gamma (already above for n80) 1 TeV JJJ vs (1 TeV) ZZ
> 
> @cbell1402 can you:
> 
> * Commit your latest code with the nXX plot creation?
> * Can you change the binning in your histograms such that the number of y bins is 100.
> * And for the n50 plots, can you set the y upper limit to 10 (from 20)
> 
> And we can focus on the samples needed for the two comparisons at the top of this message.

I have performed the Herwig showering and the files can be found on serf with "herwig" in the file name.

**Comment by jslawless** [comment]
![eeToJJJ_50k_1TeV_v_eeToZZToJJ_50k_1TeV_n90](https://user-images.githubusercontent.com/116592104/219980242-6c0071e0-58f8-4d9b-8a0f-81996a330894.png)

Got all of the technical details done. There's some weirdness that I don't understand in the where the labels end up for some of them, but I can manually set that plot by plot if we need those plots. I wonder if there's maybe something weird happening such that the xbin edges aren't consistent across all the histograms.

Anyway, now I can set details as we like, like color, font, title, spacing and such.

[split_violins.zip](https://github.com/lawrenceleejr/FragmentationStudy/files/10778190/split_violins.zip)
